server/src/main/java/com/thoughtworks/go/server/service/SecurityService.java

```diff
@@ -207,6 +209,30 @@ public boolean isAuthorizedToViewTemplates(Username username) {
         return goConfigService.cruiseConfig().isAuthorizedToViewTemplates(username.getUsername());
     }
 
+    //todo: needs refactoring use AbstractAuthenticationHelper.doesUserHasPermissions
+    //a specific method to check whether a user has permission to access agent status report has been added
+    public boolean doesUserHasPermissionsToViewAgentStatusReport(Username username, String elasticAgentProfileId) {
```

Comment:
This method needs to be removed.
Instead, use `AbstractAuthenticationHelper.doesUserHasPermissions`

As of today, `JobController` can not autowire `ApiAuthenticationHelper` component, as it is a circular dependency (Requires dependency on module 'com.thoughtworks.go.api-base.main').

To refactor this,
We will need to move the Authorization related to methods from `AbstractAuthenticationHelper` to `SecurityService`.